### PR TITLE
iconfont  transfered to iconpark

### PR DESCRIPTION
**Pull request by wuhuiluo**
## What this PR does / why we need it:
iconfont  migration to iconpark

## Does this PR introduce a user interface change?

❎ No

## ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |              |
| 🇨🇳 中文    |              |

## Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

**Comment by McDaddy** [comment]
keep title & description in English

File: shell/app/common/components/erda-icon/erda-icon.tsx [hunk 1]
```diff
@@ -80,10 +80,21 @@ export const ErdaIcon = ({ className = '', onClick, iconType, ...rest }: IProps)
...
+  | 'light-primary'
+  | 'shallow-primary'
+  | 'white'
+  | 'shallow-gray'
+  | 'black-gray'
+  | 'shallow-gray-1'
+  | 'shallo-gray-2'
+  | 'danger-red'
+  | 'shallow-gray-3'
+  | 'pink';
```
Comment (Zero-Rock) [inline shell/app/common/components/erda-icon/erda-icon.tsx:94]: type CustomColor = keyof typeof COLOR is better

File: shell/app/common/components/erda-icon/erda-icon.tsx [hunk 2]
```diff
@@ -80,10 +80,21 @@ export const ErdaIcon = ({ className = '', onClick, iconType, ...rest }: IProps)
...
 
 interface IErdaCustomIcon {
+  className?: string;
```
Comment (Zero-Rock) [inline shell/app/common/components/erda-icon/erda-icon.tsx:97]: iconpark-icon is custom element and does not support 'className',  replace with 'class'

File: shell/app/common/components/tree/edit-category.tsx [hunk 3]
```diff
@@ -82,8 +83,14 @@ export const EditCategory = (props: IProps) => {
         onPressEnter={() => handleSave()}
         onChange={(e) => setValue(e.target.value)}
       />
-      <CustomIcon className="ml12 fz18 pointer" type="duigou" onClick={handleSave} />
-      <CustomIcon className="ml12 fz18 pointer" type="close" onClick={props.onHide || handleHide} />
+      <IconCheck className="ml12 fz18 pointer" onClick={handleSave} size="16" />
```
Comment (Zero-Rock) [inline shell/app/common/components/tree/edit-category.tsx:86]: ‘fz18’ is invalid, remove it

File: shell/app/common/components/contractive-filter.scss [hunk 4]
```diff
@@ -57,6 +67,13 @@
 
     &:hover {
       color: $gray;
+      svg {
+        g {
+          path {
+            fill: red !important;
```
Comment (daskyrk) [inline shell/app/common/components/contractive-filter.scss:73]: use $red instead

File: shell/app/common/components/contractive-filter.tsx [hunk 5]
```diff
@@ -559,9 +560,12 @@ export const ContractiveFilter = ({
...
         >
           {!item.fixed && (
-            <CustomIcon
-              className="contractive-filter-item-close"
+            <ErdaCustomIcon
+              color="shallow-gray"
+              class="contractive-filter-item-close"
               type="guanbi-fill"
+              size="16"
+              style={{ fill: 'red' }}
```
Comment (daskyrk) [inline shell/app/common/components/contractive-filter.tsx:568]: Not use hard code color.

File: shell/app/common/components/contractive-filter.tsx [hunk 6]
```diff
@@ -634,9 +638,9 @@ export const ContractiveFilter = ({
             placement="bottomLeft"
           >
             <span className="contractive-filter-item">
-              <CustomIcon type="tj1" className="fz12 mr2 color-text" />
+              <IconPlus fill="rgba(0, 0, 0, 0.8)" className="fz14 mr2 mb4 color-text" />
               <span>{i18n.t('common:filter')}</span>
-              <CustomIcon type="caret-down" />
+              <IconDownOne className="hover ml2 mb2" size="12" theme="filled" fill="#bbb" />
```
Comment (daskyrk) [inline shell/app/common/components/contractive-filter.tsx:643]: Same as above

File: shell/app/common/components/erda-icon/erda-icon.tsx [hunk 7]
```diff
@@ -102,11 +113,16 @@ const COLOR = {
   white: '#ffffff',
   'light-primary': '#6a549e19', // rgba($primary, .1)
   'shallow-primary': '#6a549e99', // rgba($primary, .6)
+  'shallow-gray': '#bbb',
+  'black-gray': 'rgba(0, 0, 0, 0.85)',
+  'shallow-gray-1': 'rgba(0,0,0,0.4)',
+  'shallo-gray-2': 'rgba(0,0,0,.2)',
```
Comment (daskyrk) [inline shell/app/common/components/erda-icon/erda-icon.tsx:119]: typo

File: shell/app/common/components/erda-icon/erda-icon.tsx (hunk 1 above)
Comment (daskyrk) [reply shell/app/common/components/erda-icon/erda-icon.tsx:94]: Do we really need so many color ?

File: shell/app/modules/application/pages/build-detail/pipeline-log.tsx [hunk 8]
```diff
@@ -95,7 +95,13 @@ const PipelineLog = ({ isBuilding = false, resourceId, resourceType, className =
               : i18n.t('refresh')
           }
         >
-          <CustomIcon type="refresh" className="pointer color-text-desc" onClick={() => delayGetList(getList, 0)} />
+          <ErdaCustomIcon
+            color="black-gray"
+            size="18"
+            type="redo"
+            class="mr4 pointer color-text-desc"
```
Comment (daskyrk) [inline shell/app/modules/application/pages/build-detail/pipeline-log.tsx:102]: color and `color-text-desc` is conflict.

File: shell/app/views/index.ejs [hunk 9]
```diff
@@ -40,6 +40,8 @@
   <!-- layout -->
   <script src="https://lf1-cdn-tos.bytegoofy.com/obj/iconpark/icons_1776_8.ba89731a7541ef588c93a4ace53a6b3d.es5.js" ></script>
 
+  <script src="https://lf1-cdn-tos.bytegoofy.com/obj/iconpark/icons_138_50.88f9aabf9bddd883ad952c2742d26205.es5.js"></script>
```
Comment (daskyrk) [inline shell/app/views/index.ejs:43]: Why add a new js?

File: shell/app/yml-chart/pipeline-editor.tsx [hunk 10]
```diff
@@ -357,7 +357,7 @@ const PipelineEditor = React.forwardRef((props: IPipelineEditorProps, ref: any)
         visible={!isEmpty(errorMsg)}
         title={
           <div>
-            <CustomIcon type="guanbi-fill" className="color-danger" />
+            <ErdaCustomIcon size="16" type="guanbi-fill" fill="danger-red" class="color-danger" />
```
Comment (daskyrk) [inline shell/app/yml-chart/pipeline-editor.tsx:360]: fill is conflict with `class`

File: shell/app/views/index.ejs (hunk 9 above)
Comment (wuhuiluo) [reply shell/app/views/index.ejs:43]: change iconfont style to iconpark style

File: shell/app/common/components/erda-icon/erda-icon.tsx (hunk 1 above)
Comment (wuhuiluo) [reply shell/app/common/components/erda-icon/erda-icon.tsx:94]: iconpark property fill need

File: shell/app/common/components/contractive-filter.tsx (hunk 6 above)
Comment (wuhuiluo) [reply shell/app/common/components/contractive-filter.tsx:643]: i think only this write method

File: shell/app/common/components/erda-icon/erda-icon.tsx (hunk 7 above)
Comment (wuhuiluo) [reply shell/app/common/components/erda-icon/erda-icon.tsx:119]: iconpark property fill need

File: shell/app/common/components/contractive-filter.tsx [hunk 11]
```diff
@@ -265,7 +266,7 @@ const FilterItem = ({ itemData, value, active, onVisibleChange, onChange, onQuic
         <span className="contractive-filter-item">
           <span className="color-text-desc mr-0.5">{label}</span>
           <span className="contractive-filter-item-value nowrap">{valueText}</span>
-          <CustomIcon type="caret-down" />
+          <IconDownOne className="hover ml2 mb2" size="12" theme="filled" fill="#bbb" />
```
Comment (Zero-Rock) [inline shell/app/common/components/contractive-filter.tsx:269]: ’ml2 mb2 ‘ is deprecated className

File: shell/app/common/components/dropdown-select.tsx [hunk 12]
```diff
@@ -81,7 +80,7 @@ export const DropdownSelect = (props: IProps) => {
       {children || (
         <Button type="default" loading={loading} {...btnProps}>
           {buttonText}
-          <CustomIcon style={{ color: 'inherit' }} type="caret-down" />
+          <IconDownOne className="ml2 mb6" theme="filled" size="11" style={{ color: 'inherit' }} />
```
Comment (Zero-Rock) [inline shell/app/common/components/dropdown-select.tsx:83]: Same as above

File: shell/app/config-page/components/dropdown-select/dropdown-select.tsx [hunk 13]
```diff
@@ -130,7 +131,7 @@ const DropdownSelect = (props: CP_DROPDOWN_SELECT.Props) => {
     >
       <span className="dropdown-select-button hover-active" onClick={() => setActive(!active)}>
         {propsState?.label || label}
-        <CustomIcon style={{ color: 'inherit' }} type="caret-down" />
+        <IconDownOne style={{ color: 'inherit' }} className="caret ml2" size="14" theme="filled" />
```
Comment (Zero-Rock) [inline shell/app/config-page/components/dropdown-select/dropdown-select.tsx:134]: ’ml2 ‘ is deprecated className

File: shell/app/modules/addonPlatform/pages/common/components/addon-cards.tsx [hunk 14]
```diff
@@ -226,11 +226,13 @@ export const AddonCards = (props: IProps) => {
               <Tooltip title={projectName}>
                 <span className="nowrap project-name">
                   <CustomIcon type="overview" />
+                  123
```
Comment (Zero-Rock) [inline shell/app/modules/addonPlatform/pages/common/components/addon-cards.tsx:229]: remove debug code

File: shell/app/modules/addonPlatform/pages/common/components/addon-cards.tsx [hunk 15]
```diff
@@ -226,11 +226,13 @@ export const AddonCards = (props: IProps) => {
...
                   {projectName}
                 </span>
               </Tooltip>
               <span className="nowrap">
                 <CustomIcon type="js1" />
+                456
```
Comment (Zero-Rock) [inline shell/app/modules/addonPlatform/pages/common/components/addon-cards.tsx:235]: same as above

File: shell/app/modules/application/common/app-selector.scss [hunk 16]
```diff
@@ -26,5 +26,10 @@
     align-items: center;
     justify-items: center;
     width: 100%;
+
+    svg {
+      width: 12px !important;
+      height: 12px !important;
+    }
```
Comment (Zero-Rock) [inline shell/app/modules/application/common/app-selector.scss:33]: use the size to control the icon size

File: shell/app/modules/application/common/app-selector.tsx [hunk 17]
```diff
@@ -100,7 +101,7 @@ const headAppRender = (val: any = {}) => {
       <span className="nowrap text-base font-bold" title={name}>
         {name}
       </span>
-      <CustomIcon className="caret" type="caret-down" />
+      <IconDownOne className="caret ml4 mb4" theme="filled" />
```
Comment (Zero-Rock) [inline shell/app/modules/application/common/app-selector.tsx:104]: ’ml4 mb4 ‘ is deprecated className, use tailwind

File: shell/app/modules/application/common/components/goto-commit.tsx [hunk 18]
```diff
@@ -37,7 +37,7 @@ const GotoCommit = ({ projectId, appId, commitId = '', length = 6, className = '
         });
       }}
     >
-      <CustomIcon type="commit" />
+      <ErdaCustomIcon class="mt4" fill="pink" size="16" type="commit" />
```
Comment (Zero-Rock) [inline shell/app/modules/application/common/components/goto-commit.tsx:40]: same as above

File: shell/app/modules/application/pages/build-detail/pipeline-log.tsx [hunk 19]
```diff
@@ -95,7 +95,13 @@ const PipelineLog = ({ isBuilding = false, resourceId, resourceType, className =
               : i18n.t('refresh')
           }
         >
-          <CustomIcon type="refresh" className="pointer color-text-desc" onClick={() => delayGetList(getList, 0)} />
+          <ErdaCustomIcon
+            color="black-gray"
+            size="18"
+            type="redo"
+            class="mr4 pointer"
```
Comment (Zero-Rock) [inline shell/app/modules/application/pages/build-detail/pipeline-log.tsx:102]: same as above

File: shell/app/modules/application/pages/deploy/deploy.tsx [hunk 20]
```diff
@@ -62,7 +62,11 @@ const NewDeploy = ({ type, setCurEnv, canCreate }: IProps) => {
   return (
     <WithAuth pass={permMap[`${type.toLowerCase()}DeployOperation`].pass}>
       <div className={className} onClick={() => setCurEnv(type)}>
+<<<<<<< HEAD
         <CustomIcon type="tj1" className="text-2xl mb-4" />
+=======
+        <ErdaCustomIcon fill="shallow-gray-4" type="tj1" class="fz24 mb16 hover" />
+>>>>>>> 1ccf5866... feat: iconfont transfered to iconpark
```
Comment (Zero-Rock) [inline shell/app/modules/application/pages/deploy/deploy.tsx:69]: conflict

File: shell/app/modules/application/pages/deploy/runtime-box.tsx [hunk 21]
```diff
@@ -148,7 +148,9 @@ const RuntimeBox = (props: IProps) => {
       <div className="flex justify-between items-center runtime-box">
         <div className="flex justify-between items-center runtime-box-header">
           <div className="branch disabled">
-            <CustomIcon type="slbb" />
+            <div style={{ margin: '0 4px 0 0' }}>
```
Comment (Zero-Rock) [inline shell/app/modules/application/pages/deploy/runtime-box.tsx:151]: don't use Inline style，replace with className="mt-1"

File: shell/app/modules/application/pages/deploy/runtime-box.tsx [hunk 22]
```diff
@@ -167,7 +169,9 @@ const RuntimeBox = (props: IProps) => {
       >
         <div className="flex justify-between items-center runtime-box-header">
           <div className="branch">
-            <CustomIcon type="slbb" />
+            <div style={{ margin: '0 4px 0 0' }}>
```
Comment (Zero-Rock) [inline shell/app/modules/application/pages/deploy/runtime-box.tsx:172]: same as above

File: shell/app/modules/application/pages/deploy/runtime-box.tsx [hunk 23]
```diff
@@ -187,7 +191,7 @@ const RuntimeBox = (props: IProps) => {
           <div>
             <Tooltip title={i18n.t('application:view version information')}>
               <span className="text-link release-link" onClick={(e) => gotoRelease(releaseId, e)}>
-                <CustomIcon type="bb" />
+                <ErdaCustomIcon fill="pink" width="20" height="21" type="bb" />
```
Comment (Zero-Rock) [inline shell/app/modules/application/pages/deploy/runtime-box.tsx:194]: what is bb

File: shell/app/modules/project/pages/test-manage/case/manual-test.tsx [hunk 24]
```diff
@@ -186,7 +188,7 @@ const ManualTest = () => {
               prefix={<IconSearch />}
             />
             <Button onClick={() => setEnhanceFilterVisible(true)}>
-              <CustomIcon type="filter" />
+              <ErdaCustomIcon stroke="black-gray" class="mt2" width="16" height="18" type="filter" />
```
Comment (Zero-Rock) [inline shell/app/modules/project/pages/test-manage/case/manual-test.tsx:191]: ’mt2 ‘ ar deprecated className

File: shell/app/common/components/erda-icon/erda-icon.tsx [hunk 25]
```diff
@@ -101,6 +102,15 @@ const COLOR = {
...
+  'shallow-gray-3': 'rgba(0,0,0,.8)',
+  'shallow-gray-5': 'rgba(0,0,0,0.5)',
+  'shallow-gray-6': 'rgba(0,0,0,0.6)',
+  'danger-red': '#df3409',
+  pink: 'rgba(106, 84, 158, 0.8)',
```
Comment (Zero-Rock) [inline shell/app/common/components/erda-icon/erda-icon.tsx:113]: add a new prop "opacity", the color is determined by the basic color and opacity

File: shell/app/modules/application/pages/deploy/runtime-box.tsx (hunk 23 above)
Comment (wuhuiluo) [reply shell/app/modules/application/pages/deploy/runtime-box.tsx:194]: iconpark type name

File: shell/app/common/components/dropdown-select.tsx [hunk 26]
```diff
@@ -81,7 +80,7 @@ export const DropdownSelect = (props: IProps) => {
       {children || (
         <Button type="default" loading={loading} {...btnProps}>
           {buttonText}
-          <CustomIcon style={{ color: 'inherit' }} type="caret-down" />
+          <IconDownOne className="ml-0.5 mb-1.5" theme="filled" size="11" style={{ color: 'inherit' }} />
```
Comment (Zero-Rock) [inline shell/app/common/components/dropdown-select.tsx:83]: remove color

File: shell/app/common/components/tree/edit-category.tsx [hunk 27]
```diff
@@ -108,7 +115,7 @@ export const EditCategory = (props: IProps) => {
     >
       {createButton || (
         <Button type="primary" onClick={() => value && onSubmit({ name: value })}>
-          <CustomIcon type="cir-add" className="mr4" />
+          <ErdaCustomIcon type="cir-add" class="mr4" size="16" fill="white" />
```
Comment (Zero-Rock) [inline shell/app/common/components/tree/edit-category.tsx:118]: replace mr4

File: shell/app/modules/application/pages/settings/components/app-notify/common-notify-group.tsx [hunk 28]
```diff
@@ -148,7 +148,7 @@ export const ListTargets = ({
       text = `${i18n.t('application:notify role')}：${map(values, (obj) => roleMap[obj.receiver]).join(',')}`;
       targetsEle = (
         <>
-          <CustomIcon type="sidebarUser" className="text-desc" />
+          <ErdaCustomIcon fill="shallow-gray-1" size="16" type="sidebarUser" class="color-text-desc mr-1" />
```
Comment (Zero-Rock) [inline shell/app/modules/application/pages/settings/components/app-notify/common-notify-group.tsx:151]: remove text-desc

File: shell/app/common/components/erda-icon/erda-icon.tsx [hunk 29]
```diff
@@ -94,16 +96,37 @@ interface IErdaCustomIcon {
...
 
 const COLOR = {
-  primary: '#6a549e',
-  white: '#ffffff',
-  'light-primary': '#6a549e19', // rgba($primary, .1)
-  'shallow-primary': '#6a549e99', // rgba($primary, .6)
+  primary: '106,84,158',
+  white: '255,255,255',
+  'shallow-gray': '187,187,187',
+  'opacity-gray': '0,0,0',
```
Comment (Zero-Rock) [inline shell/app/common/components/erda-icon/erda-icon.tsx:106]: 0,0,0 is black

File: shell/app/modules/application/pages/repo/repo-branch.scss [hunk 30]
```diff
@@ -14,4 +14,8 @@
       line-height: 24px;
     }
   }
+
+  .branch-item-right {
```
Comment (Zero-Rock) [inline shell/app/modules/application/pages/repo/repo-branch.scss:18]: use tailwind

**Pull request merged**